tools/catsitemap.py

```diff
@@ -0,0 +1,119 @@
+"""Script for automatic generation/download of sitemaps for pyansys projects.
+
+Intended for the pyansys project and to be used with the update-gh-pages
+workflow.
+
+"""
+
+from pathlib import Path
+from xml.dom import minidom
+import xml.etree.ElementTree as ET
+
+from links import LINKS
+import requests
+
+
+def download_file(url: str, dest_path: Path) -> None:
+    """Given a sitemap url, this function downloads the file into (dest_path).
+
+    Parameters
+    ----------
+    url : str
+        The url of the sitemap file to be downloaded
+    dest_path : Path
+        The destination path to save the downloaded file
+
+    Raises
+    ------
+    requests.exceptions.Timeout
+        Raises this exception when accessing a link takes too long
+    """
+    # Send the request
+    try:
+        response = requests.get(url, stream=True, timeout=30)
+    except requests.exceptions.Timeout as e:
+        print(f"Timed out while trying to get download the sitemap at this url: {url}")
+        raise e
+
+    # Write the file content to the specified location
+    with open(dest_path, mode="wb") as file:
```

Comment:
We are not dealing with binary data. Therefore, no need for the `b` here. Since ``dest_path`` is a Path object, you can also do:

```suggestion
    with dest_path.open(mode="w", encoding="utf-8") as file:
```

Reply:
I ended up changing the syntax here to use the text attribute of the response object for writing to the created file.